tests/test_scalars.py

```diff
@@ -15,26 +22,568 @@
+    check(assert_type(ts <= c_datetimeindex, np_ndarray_bool), np.ndarray)
+
+    check(assert_type(ts > c_np_ndarray_dt64, np_ndarray_bool), np.ndarray)
+    check(assert_type(ts <= c_np_ndarray_dt64, np_ndarray_bool), np.ndarray)
+
+    check(assert_type(ts > c_series_timestamp, "pd.Series[bool]"), pd.Series, bool)
+    check(assert_type(ts <= c_series_timestamp, "pd.Series[bool]"), pd.Series, bool)
+
+    check(assert_type(ts > c_series_dt64, "pd.Series[bool]"), pd.Series, bool)
+    check(assert_type(ts <= c_series_dt64, "pd.Series[bool]"), pd.Series, bool)
+
+    check(assert_type(c_np_dt64 > ts, Any), np.bool_)
+    check(assert_type(c_np_dt64 <= ts, Any), np.bool_)
+
+    gt = check(assert_type(c_dt_datetime > ts, bool), bool)
+    lte = check(assert_type(c_dt_datetime <= ts, bool), bool)
+    assert gt != lte
+
+    check(assert_type(c_datetimeindex > ts, np_ndarray_bool), np.ndarray)
+    check(assert_type(c_datetimeindex <= ts, np_ndarray_bool), np.ndarray)
+
+    check(assert_type(c_np_ndarray_dt64 > ts, np_ndarray_bool), np.ndarray)
+    check(assert_type(c_np_ndarray_dt64 <= ts, np_ndarray_bool), np.ndarray)
+
+    check(assert_type(c_series_dt64 > ts, "pd.Series[bool]"), pd.Series, bool)
+    check(assert_type(c_series_dt64 <= ts, "pd.Series[bool]"), pd.Series, bool)
+
+    gte = check(assert_type(ts >= c_timestamp, bool), bool)
+    lt = check(assert_type(ts < c_timestamp, bool), bool)
+    assert gte != lt
+
+    gte = check(assert_type(ts >= c_np_dt64, bool), bool)
+    lt = check(assert_type(ts < c_np_dt64, bool), bool)
+    assert gte != lt
+
+    gte = check(assert_type(ts >= c_dt_datetime, bool), bool)
+    lt = check(assert_type(ts < c_dt_datetime, bool), bool)
+    assert gte != lt
+
+    check(assert_type(ts >= c_datetimeindex, np_ndarray_bool), np.ndarray)
+    check(assert_type(ts < c_datetimeindex, np_ndarray_bool), np.ndarray)
+
+    check(assert_type(ts >= c_np_ndarray_dt64, np_ndarray_bool), np.ndarray)
+    check(assert_type(ts < c_np_ndarray_dt64, np_ndarray_bool), np.ndarray)
+
+    check(assert_type(ts >= c_series_timestamp, "pd.Series[bool]"), pd.Series, bool)
+    check(assert_type(ts < c_series_timestamp, "pd.Series[bool]"), pd.Series, bool)
+
+    check(assert_type(ts >= c_series_dt64, "pd.Series[bool]"), pd.Series, bool)
+    check(assert_type(ts < c_series_dt64, "pd.Series[bool]"), pd.Series, bool)
+
+    gte = check(assert_type(c_dt_datetime >= ts, bool), bool)
+    lt = check(assert_type(c_dt_datetime < ts, bool), bool)
+    assert gte != lt
+
+    check(assert_type(c_np_dt64 >= ts, Any), np.bool_)
+    check(assert_type(c_np_dt64 < ts, Any), np.bool_)
+
+    check(assert_type(c_datetimeindex >= ts, np_ndarray_bool), np.ndarray)
+    check(assert_type(c_datetimeindex < ts, np_ndarray_bool), np.ndarray)
+
+    check(assert_type(c_np_ndarray_dt64 >= ts, np_ndarray_bool), np.ndarray)
+    check(assert_type(c_np_ndarray_dt64 < ts, np_ndarray_bool), np.ndarray)
```

Comment:
same comment about additional argument to `check`

Reply:
added